SurveyorCore/src/main/java/org/wwarn/surveyor/client/model/FilterConfig.java

```diff
@@ -54,18 +51,25 @@ public String getFilterLabel() {
 
     private String filterLabel;
 
+    public void addFilterSetting(FilterSetting filterSetting){
+        filters.add(filterSetting);
+    }
+
+    @Deprecated
     public void addFilter(String filterColumn, String filterFieldName, String filterFieldLabel, HashMap<String, String> filterFieldValueToLabelMap) {
-        filters.add(new FilterSetting.FilterSettingsBuilder(filterColumn, filterFieldName, filterFieldLabel).setFilterFieldValueToLabelMap(filterFieldValueToLabelMap).build());
+        addFilterSetting(new FilterSetting.FilterSettingsBuilder(filterColumn, filterFieldName, filterFieldLabel).setFilterFieldValueToLabelMap(filterFieldValueToLabelMap).build());
```

Comment:
It is an automatic refactor consequence, doesn't matter really if the code is updated.